hi-ml/src/health_ml/utils/regression_test_utils.py

```diff
@@ -0,0 +1,169 @@
+#  -------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  -------------------------------------------------------------------------------------------
+import logging
+import os
+from pathlib import Path
+import shutil
+import tempfile
+from typing import List, Optional
+
+from azureml.core import Run
+import pandas as pd
+
+from health_azure.utils import PARENT_RUN_CONTEXT, RUN_CONTEXT, is_running_in_azure_ml
+
+
+REGRESSION_TEST_OUTPUT_FOLDER = "OUTPUT"
+REGRESSION_TEST_AZUREML_FOLDER = "AZUREML_OUTPUT"
+REGRESSION_TEST_AZUREML_PARENT_FOLDER = "AZUREML_PARENT_OUTPUT"
+CONTENTS_MISMATCH = "Contents mismatch"
+FILE_FORMAT_ERROR = "File format error"
+MISSING_FILE = "Missing"
+CSV_SUFFIX = ".csv"
+TEXT_FILE_SUFFIXES = [".txt", ".json", ".html", ".md"]
+
+
+def compare_files(expected: Path, actual: Path, csv_relative_tolerance: float = 0.0) -> str:
+    """
+    Compares two files for regression testing. Returns an empty string if they are identical.
+    Otherwise, an error message with details is returned. This handles known text file formats,
+    and differences in line breaks are ignored. All other files are treated as binary, and
+    compared on a byte-by-byte basis.
+
+    :param expected: A file that contains the expected contents. The type of comparison (text or binary) is chosen
+    based on the extension of this file.
+    :param actual: A file to be checked against the expected file.
+    :param csv_relative_tolerance: When comparing CSV files, maximum allowed relative discrepancy.
+    If 0.0, do not allow any discrepancy.
+    :return: An empty string if the files are identical, an error message with details otherwise.
+    """
+    def print_lines(prefix: str, lines: List[str]) -> None:
+        num_lines = len(lines)
+        count = min(5, num_lines)
+        logging.info(f"{prefix} {num_lines} lines, first {count} of those:")
+        logging.info(os.linesep.join(lines[:count]))
+
+    def try_read_csv(prefix: str, file: Path) -> Optional[pd.DataFrame]:
+        try:
+            return pd.read_csv(file)
+        except Exception as ex:
```

Comment:
Can we catch a more specific exception here?

Reply:
I'm deliberately generous here, because I want all errors to be collected _before_ the program terminates because it sees discrepancies. I've changed the logging here though to print exception details as a warning.